### PR TITLE
Json configs

### PR DESCRIPTION
**Pull request by sp0lsh**

### CODE REVIEW

File: PolyEngine/Engine/Src/CoreConfig.hpp [hunk 1]
```diff
@@ -11,6 +22,19 @@ namespace Poly
...
+
+		bool GetIsLoadedFromFile() const;
+
+		bool ReloadFromFile();
+
+	private:
+
+		bool IsLoadedFromFile = false;
+
+		std::unordered_map<eResourceSource, Dynarray<String>> AssetsPaths;
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: Please change it to `EnumArray`

File: PolyEngine/Standalone/Windows/CoreConfig.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+	"DebugNormalsFlag": false,
```
Comment (MuniuDev) [inline PolyEngine/Standalone/Windows/CoreConfig.json]: Please remove other config related stuf other than asset paths and rename file to AssetPaths.json

File: PolyEngine/Engine/Src/CoreConfig.hpp [hunk 3]
```diff
@@ -11,6 +22,19 @@ namespace Poly
 		bool DebugNormalsFlag = false;
 		bool WireframeRendering = false;
 		bool DisplayFPS = true;
+
+		Dynarray<String> GetAssetsPaths(eResourceSource Source) const;
+
+		bool GetIsLoadedFromFile() const;
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: If getter returns bool it's nicer to name it starting with "Is". "GetIs" just looks akward.

File: PolyEngine/Engine/Src/ResourceManager.hpp [hunk 4]
```diff
@@ -36,9 +38,9 @@ namespace Poly
 	{
 	public:
 		//------------------------------------------------------------------------------
-		static T* Load(const String& relativePath, bool absolute = true)
+		static T* Load(const String& path, bool isAbsolute = true, eResourceSource source = eResourceSource::ENGINE)
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/ResourceManager.hpp]: Maybe we can remove `isAbsolute` and replace it with `eResourceSource::ABSOLUTE` ? Then the name of the function `LoadTextFileRelative` should be changed to something else, i.e.: `LoadTextFileResource`.

File: PolyEngine/Core/Src/FileIO.hpp [hunk 5]
```diff
@@ -55,4 +55,18 @@ namespace Poly {
 		else
 			throw FileIOException("File save failed");
 	}
+
+	inline bool FileExists(const String& path)
+	{
+		FILE *f;
+		fopen_s(&f, path.GetCStr(), "r");
+		if (f) {
```
Comment (MuniuDev) [inline PolyEngine/Core/Src/FileIO.hpp]: Please move the brace to new line.

File: PolyEngine/Engine/Src/CoreConfig.hpp (hunk 3 above)
Comment (sp0lsh) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: ok, it was workaround for property named the same. What is convention in such situation?

Comment (MuniuDev) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: Property can be simply named `LoadedFromFile`

Comment (sp0lsh) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: done

File: PolyEngine/Engine/Src/CoreConfig.hpp (hunk 1 above)
Comment (sp0lsh) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: done

File: PolyEngine/Standalone/Windows/CoreConfig.json (hunk 2 above)
Comment (sp0lsh) [inline PolyEngine/Standalone/Windows/CoreConfig.json]: done

File: PolyEngine/Core/Src/FileIO.hpp (hunk 5 above)
Comment (sp0lsh) [inline PolyEngine/Core/Src/FileIO.hpp]: done

File: PolyEngine/Engine/Src/ResourceManager.hpp (hunk 4 above)
Comment (sp0lsh) [inline PolyEngine/Engine/Src/ResourceManager.hpp]: good idea but I would leave this to next iteration on asset loading

**Pull request merged**